specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1097 @@
+Channellable duck type. If the referenced `spec.channel` does not currently
+exist or its `Ready` condition is not `true`, then the Subscription's `Ready`
+condition MUST NOT be `true`, and the reason SHOULD indicate that the
+corresponding channel is missing or not ready.
+
+The Subscription MUST also set the `status.physicalSubscription` URIs by
+resolving the `spec.subscriber`, `spec.reply`, and
+`spec.delivery.deadLetterSink` as described in
+[Addressable resolution](#addressable-resolution) before setting the `Ready`
+condition to `true`. If any of the addressable fields fails resolution, the
+Subscription MUST set the `Ready` condition to `false`, and at least one
+condition should indicate the reason for the error. (It is acceptable for none
+of the `spec.subscriber`, `spec.reply`, and `spec.delivery.deadLetterSink`
+fields to contain a `ref` field.)
+
+<!-- TODO: is at least one of `subscriber` and `reply` required? -->
+
+Once created, the Subscription's `spec.channel` SHOULD NOT permit updates; to
+change the `spec.channel`, the Subscription can be deleted and re-created. This
+pattern is chosen to make it clear that changing the `spec.channel` is not an
+atomic operation, as it may span multiple storage systems. Changes to
+`spec.subscriber`, `spec.reply`, `spec.delivery` and other fileds SHOULD be
+permitted, as these could occur within a single storage system.
+
+When a Subscription becomes associated with a channel (either due to creating
+the Subscription or the channel), the Subscription MUST only set the `Ready`
+condition to `true` after the channel has been configured to send all future
+events to the Subscriptions `spec.subscriber`. The Channel MAY send some events
+to the Subscription before prior to the Subscription's `Ready` condition being
+set to `true`. When a Subscription is deleted, the Channel MAY send some
+additional events to the Subscription's `spec.subscriber`.
+
+<!--
+TODO: channel-compatible CRDs (Channelable)
+-->
+
+## Addressable Resolution
+
+Both Trigger and Subscription have optional object references (`ref` in
+`spec.subscriber`, `spec.delivery.deadLetterSink`, and `spec.reply` for
+Subscription) which are expected to conform to the Addressable partial schema
+("duck type"). An object conforms with the Addressable partial schema if it
+contains a `status.address.url` field containing a URL which may be used to
+deliver CloudEvents over HTTP. As a special case, Kubernetes `v1` `Service`
+objects are considered to have a `status.address.url` of
+`http://<service-dns-name>/`. If one of these object references points to an
+object which does not currently satisfy this partial schema (either because the
+`status.address.url` field is empty, or because the object does not have that
+field), then the Trigger or Subscription MUST indicate an error by setting the
+`Ready` condition to `false`, and SHOULD include an indication of the error in a
+condition reason or type.
```

Comment:
Since the destination is the ref+uri, wonder if this might be the place to describe that.
Ref => resolve as you say
Uri => fully specified, for example to external resources (that you mentioned in your other doc)
Ref+Uri => Ref resolved as above, Uri is relative URL then to the result of Ref resolving.